### PR TITLE
[archetype] Minor changes to fulfill static-code-analysis tool

### PR DESCRIPTION
**Pull request by cweitkamp**
- Minor changes to fulfill static-code-analysis tool
- Applied code formatter
- Corrected typos
- Removed not resolvable import `org.hamcrest.collections` from `MANIFEST.MF`. Build failed because of it (see also openhab/openhab2-addons#3157).
- Added template for license file
```
[ERROR] Cannot resolve project dependencies:
[ERROR]   Software being installed: org.eclipse.smarthome.binding.mytest.test 0.10.0.qualifier
[ERROR]   Missing requirement: org.eclipse.smarthome.binding.mytest.test 0.10.0.qualifier requires 'package org.hamcrest.collections 0.0.0' but it could not be found
```
- Added dependency to `pom.xml`
- Applied POM Code Convention

One question: We included the `NOTICE` file in the `build.properties` but it doesn't exist in the archetype resources. Shall we add it? Or do we have to copy it from somewhere else? What is the suggested way to go?

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

**Comment by wborn** [comment]
Can you also convert the Windows to Unix line endings in them MANIFEST.MF files @cweitkamp? :-)

File: tools/archetype/binding/src/main/resources/archetype-resources/OSGI-INF/.gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
-*.xml
\ No newline at end of file
+*.xml
+
```
Comment (htreu) [inline tools/archetype/binding/src/main/resources/archetype-resources/OSGI-INF/.gitignore]: one newline too much

**Comment by cweitkamp** [comment]
@wborn @htreu Thanks for review, I incorporated your suggestions.

File: tools/archetype/binding/src/main/resources/archetype-resources/src/main/java/internal/__bindingIdCamelCase__HandlerFactory.java [hunk 2]
```diff
@@ -40,7 +40,7 @@
  *
  * @author ${author} - Initial contribution
  */
-@Component(service = ThingHandlerFactory.class, immediate = true, configurationPid = "binding.${bindingId}")
+@Component(service = ThingHandlerFactory.class, immediate = true)
```
Comment (triller-telekom) [inline tools/archetype/binding/src/main/resources/archetype-resources/src/main/java/internal/__bindingIdCamelCase__HandlerFactory.java]: I thought we added the `configurationPid` in order to make the binding appear in the PaperUI binding configuration screen, or as I mistaken here?

Comment (cweitkamp) [reply tools/archetype/binding/src/main/resources/archetype-resources/src/main/java/internal/__bindingIdCamelCase__HandlerFactory.java]: I will revert this change. Makes sense.

I am just asking myself why the ESH bindings do not use the `configurationPid` at the moment.

> What was it that the SAT tool complained about?

In ESH only code format and newline in `.gitignore`. But
> We included the `NOTICE` file in the `build.properties` but it doesn't exist in the archetype resources. Shall we add it? Or do we have to copy it from somewhere else? What is the suggested way to go?

In OH2 additionally
```
[ERROR] Code Analysis Tool has found: 
 4 error(s)! 
 3 warning(s) 
 4 info(s)
[ERROR] org.openhab.binding.mytestbinding.internal.MyTestBindingHandlerFactory.java:[2]
Header line doesn't match pattern ^ \* Copyright \(c\) 2010-2018 by the respective copyright holders\.$
[ERROR] org.openhab.binding.mytestbinding.internal.MyTestBindingConfiguration.java:[2]
Header line doesn't match pattern ^ \* Copyright \(c\) 2010-2018 by the respective copyright holders\.$
[ERROR] org.openhab.binding.mytestbinding.handler.MyTestBindingHandler.java:[2]
Header line doesn't match pattern ^ \* Copyright \(c\) 2010-2018 by the respective copyright holders\.$
[ERROR] org.openhab.binding.mytestbinding.MyTestBindingBindingConstants.java:[2]
Header line doesn't match pattern ^ \* Copyright \(c\) 2010-2018 by the respective copyright holders\.$
```
Steps to fix it in OH2
- replace `NOTICE` with `about.html` in `build-properties`
- run `mvn license:format`

File: tools/archetype/binding/src/main/resources/archetype-resources/META-INF/MANIFEST.MF [hunk 3]
```diff
@@ -1,26 +1,26 @@
...
+Manifest-Version: 1.0
+Bundle-ActivationPolicy: lazy
+Bundle-ClassPath: .
+Bundle-ManifestVersion: 2
+Bundle-Name: ${bindingIdCamelCase} Binding
+Bundle-RequiredExecutionEnvironment: JavaSE-1.8
+Bundle-SymbolicName: ${artifactId};singleton:=true
+Bundle-Vendor: ${vendorName}
+Bundle-Version: ${version.replaceAll("-SNAPSHOT", ".qualifier")}
+Export-Package:
```
Comment (wborn) [inline tools/archetype/binding/src/main/resources/archetype-resources/META-INF/MANIFEST.MF]: Can you readd the space at the end of this line?
Without the space the build fails with: `Error assembling JAR: invalid header field`

File: tools/archetype/binding/src/main/resources/archetype-resources/META-INF/MANIFEST.MF [hunk 4]
```diff
@@ -1,26 +1,26 @@
...
+ ${package},
+ ${package}.handler
+Import-Package:
```
Comment (wborn) [inline tools/archetype/binding/src/main/resources/archetype-resources/META-INF/MANIFEST.MF]: Can you readd the space at the end of this line?
Without the space the build fails with: `Error assembling JAR: invalid header field`

File: tools/archetype/binding.test/src/main/resources/archetype-resources/META-INF/MANIFEST.MF [hunk 5]
```diff
@@ -1,31 +1,30 @@
...
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: ${bindingIdCamelCase} Binding Tests
+Bundle-RequiredExecutionEnvironment: JavaSE-1.8
+Bundle-SymbolicName: ${artifactId};singleton:=true
+Bundle-Vendor: ${vendorName}
+Bundle-Version: ${version.replaceAll("-SNAPSHOT", ".qualifier")}
+Export-Package: ${package};uses:="org.eclipse.smarthome.test"
+Fragment-Host: ${package}
+Import-Package:
```
Comment (wborn) [inline tools/archetype/binding.test/src/main/resources/archetype-resources/META-INF/MANIFEST.MF]: Can you readd the space at the end of this line?
Without the space the build fails with: `Error assembling JAR: invalid header field`

**Comment by cweitkamp** [comment]
@wborn Done.

File: tools/archetype/binding/src/main/resources/archetype-resources/ESH-INF/binding/binding.xml [hunk 6]
```diff
@@ -1,8 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<binding:binding id="${bindingId}"
-				 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-				 xmlns:binding="http://eclipse.org/smarthome/schemas/binding/v1.0.0"
-				 xsi:schemaLocation="http://eclipse.org/smarthome/schemas/binding/v1.0.0 http://eclipse.org/smarthome/schemas/binding-1.0.0.xsd">
+<binding:binding id="mytestbinding" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (wborn) [inline tools/archetype/binding/src/main/resources/archetype-resources/ESH-INF/binding/binding.xml]: The `id="${bindingId}"` got substituted :-)

Comment (cweitkamp) [reply tools/archetype/binding/src/main/resources/archetype-resources/ESH-INF/binding/binding.xml]: Ups, sloppy mistake. Fixed it. Thanks for the hint.

**Pull request merged**